pullbug/gitlab_bug.py

```diff
@@ -76,7 +76,30 @@ def iterate_merge_requests(cls, merge_requests, wip):
         return final_message
 
     @classmethod
-    def prepare_message(cls, merge_request):
+    def get_projects(cls, gitlab_scope, gitlab_state):
+        """Get all projects of the GITLAB_API_URL.
+        """
+        LOGGER.info('Bugging GitLab for projects...')
+        try:
+            response = requests.get(
+                f"{GITLAB_API_URL}/projects?scope={gitlab_scope}&state={gitlab_state}&per_page=100",
+                headers=GITLAB_HEADERS
+            )
+            print(response.json())
+            LOGGER.info('GitLab projects retrieved!')
+            if 'does not have a valid value' in response.text:
+                error = f'Could not retrieve GitLab projects due to bad parameter: {gitlab_scope} | {gitlab_state}.'  # noqa
+                LOGGER.error(error)
+                raise ValueError(error)
+        except requests.exceptions.RequestException as response_error:
+            LOGGER.error(
+                f'Could not retrieve GitLab projects: {response_error}'
+            )
+            raise requests.exceptions.RequestException(response_error)
+        return response.json()
```

Comment:
Although this probably works, it adds an extra API call to the mix and a bit of extra complexity. A better approach may be to infer the repo name from a property in the merge_request response such as the URL or reference. This way we get it directly from the source and don't need an extra API call.